### DISCUSSION TITLE
[V4] automatic namespace detection

### DISCUSSION THREAD

**Discussion by The-Code-Monkey** [tailwindlabs/tailwindcss#13420] [Ideas]
Hey, I've been testing out V4 and I'm impressed the only thing I've been a little confused about is the theme function instead of the apply directive, I get the reason for the switch but I don't find it any faster than typing out the CSS variable that tailwind adds, is it possible instead of writing `border-radius: theme(borderRadius.lg)`

To not be able to type `border-radius: theme(lg)`

And grab the namespace of border radius from the CSS that's just been read. Otherwise I might as well just write `border-radius: var(--borderRadius-lg);` as that is the same thing

**Comment by The-Code-Monkey** [comment]
Obviously if no value is found it should fall back to the current implementation

**Comment by The-Code-Monkey** [comment]
Bump

**Comment by adamwathan** [comment]
Hey! Currently v4 doesn't actually support `theme` but it does support `@apply` as well as `var(--radius-lg)`. The `theme` function was never meant to be less typing than `var(...)`, it only existed because historically we didn't generate the variables. The automatic namespace detection idea is neat but I think a bit more clever/magical than I want to be with things going forward at the moment — not saying definitely no forever but no plans to add it at the moment. Thanks for the idea regardless!

**Reply by The-Code-Monkey** [reply]
Awesome cheers for the reply.
